### PR TITLE
[dtensor][random] use torch.uint64 as the seed/offset tensor dtype to avoid overflow

### PR DESCRIPTION
**Pull request by XilunWu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #141532
* #141223
* #141220
* #141731

**Summary**
DTensor RNG code raises error if the seed passed in is beyong `torch.int64` range (e.g. `torch.tensor([2**64-1])` raises error). The solution is to specify the `dtype=torch.uint64` in the `torch.tensor()` call.

cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

**Review by wconstab — APPROVED** [review]
glad you found this, i noticed the error once but not consistently, and didn't track down where it came from.

File: torch/distributed/tensor/_random.py [hunk 1]
```diff
@@ -198,7 +198,7 @@ def set_offset(self, name: str, offset: int) -> None:
             )
 
         seed_tensor = (self.rng_states[name])[0:8]
-        offset_tensor = torch.tensor([offset]).view(torch.uint8)
+        offset_tensor = torch.tensor([offset], dtype=torch.uint64).view(torch.uint8)
```
Comment (Skylion007) [inline torch/distributed/tensor/_random.py:201]: Isn't this immediately cast to uint8?

**Review by Skylion007 — COMMENTED** [review]
Bit confused here

**Comment by Skylion007** [comment]
Oh I see, this is just to hide an error.

File: torch/distributed/tensor/_random.py (hunk 1 above)
Comment (wconstab) [reply torch/distributed/tensor/_random.py:201]: offset comes in as a uint64 essentially.  we're using a tensor to convert it to its constituent bytes because this is the format used by torch's RNG state.  the initial dtype=uint64 is needed to prevent overflow warnings mainly.
